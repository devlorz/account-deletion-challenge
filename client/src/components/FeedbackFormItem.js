import PropTypes from 'prop-types'
import React from 'react'

import InputForm from './InputForm'

const FeedbackFormItem = ({ item, isChecked, onCheckBoxClick }) => (
  <div>
    <label>
      <input
        type="checkbox"
        defaultChecked={isChecked}
        onClick={onCheckBoxClick}
      />
      {item.title}
    </label>
    <InputForm
      stack={item.stack}
      canComment={item.canComment}
      placeHolder={item.placeHolder}
      isRender={isChecked}
    />
  </div>
)

FeedbackFormItem.PropTypes = {
  item: PropTypes.object,
  isChecked: PropTypes.bool,
  onCheckBoxClick: PropTypes.func,
}

export default FeedbackFormItem
