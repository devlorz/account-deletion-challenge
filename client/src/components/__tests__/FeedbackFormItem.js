import React from 'react'
import { render } from 'react-testing-library'

import FeedbackFormItem from '../FeedbackFormItem'

test('snapshot', () => {
  const { container } = render(
    <FeedbackFormItem
      item={{
        canComment: true,
        placeHolder: 'Please specify',
        stack: 'others',
        title: 'Other reason(s)...',
      }}
      isChecked={true}
      onCheckBoxClick={jest.fn()}
    />
  )
  expect(container.firstChild).toMatchSnapshot()
})
