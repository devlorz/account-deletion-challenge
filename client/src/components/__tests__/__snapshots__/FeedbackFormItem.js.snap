// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  data-reactroot=""
>
  <label>
    <input
      type="checkbox"
      value="on"
    />
    <!-- react-text: 4 -->
    Other reason(s)...
    <!-- /react-text -->
  </label>
  <div>
    <input
      name="others"
      placeholder="Please specify"
      type="text"
    />
  </div>
</div>
`;
